tests/components/deconz/test_deconz_event.py

```diff
@@ -67,53 +67,40 @@ async def test_deconz_events(hass):
     switch_2_battery_level = hass.states.get("sensor.switch_2_battery_level")
     assert switch_2_battery_level.state == "100"
 
-    mock_listener = Mock()
-    unsub = hass.bus.async_listen(CONF_DECONZ_EVENT, mock_listener)
+    events = async_capture_events(hass, CONF_DECONZ_EVENT)
 
     gateway.api.sensors["1"].update({"state": {"buttonevent": 2000}})
     await hass.async_block_till_done()
 
-    assert len(mock_listener.mock_calls) == 1
-    assert mock_listener.mock_calls[0][1][0].data == {
+    assert len(events) == 1
+    assert events[0].data == {
         "id": "switch_1",
         "unique_id": "00:00:00:00:00:00:00:01",
         "event": 2000,
     }
 
-    unsub()
-
-    mock_listener = Mock()
-    unsub = hass.bus.async_listen(CONF_DECONZ_EVENT, mock_listener)
-
     gateway.api.sensors["3"].update({"state": {"buttonevent": 2000}})
     await hass.async_block_till_done()
 
-    assert len(mock_listener.mock_calls) == 1
-    assert mock_listener.mock_calls[0][1][0].data == {
+    assert len(events) == 2
+    assert events[1].data == {
         "id": "switch_3",
         "unique_id": "00:00:00:00:00:00:00:03",
         "event": 2000,
         "gesture": 1,
     }
 
-    unsub()
-
-    mock_listener = Mock()
-    unsub = hass.bus.async_listen(CONF_DECONZ_EVENT, mock_listener)
-
     gateway.api.sensors["4"].update({"state": {"gesture": 0}})
     await hass.async_block_till_done()
 
-    assert len(mock_listener.mock_calls) == 1
-    assert mock_listener.mock_calls[0][1][0].data == {
+    assert len(events) == 3
+    assert events[2].data == {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5451397)